### PR TITLE
Invite New Users through Email

### PR DESCRIPTION
**Pull request by itsliterallymonique**
### 🦋 Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

### 🦧 Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 

## 🐸 How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

## 🐙 Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### 🐁 Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @me-liu

### CODE REVIEW

File: api/supabase/queries/auth.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import supabase from "@/api/supabase/createClient";
+
+export async function signInWithEmail(email: string) {
```
Comment (me-liu) [inline api/supabase/queries/auth.ts]: actuallyy can you rename to signInWithMagicLink?

File: api/supabase/queries/auth.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      shouldCreateUser: true,
+      emailRedirectTo: "http://localhost:3000/auth/login",
+    },
+  });
+
+  if (error) {
+    console.error("Error sending email:", error.message);
+  }
+}
+
```
Comment (me-liu) [inline api/supabase/queries/auth.ts]: can you add a comment here that's like "returns True if there is an unaccepted invite"

File: api/supabase/queries/auth.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+export async function checkInvites(invite_id: UUID) {
+  const { data, error } = await supabase
+    .from("invite")
+    .select("*")
+    .eq("invite_id", invite_id)
+    .single();
+  if (error) {
+    console.error("Error fetching invite from invite_id:", error.message);
+  }
+  const invite_data: Invite = data as Invite;
```
Comment (me-liu) [inline api/supabase/queries/auth.ts]: add  
`if (!data) return false;
`
before pending invite check
